src/input_xml.F90

```diff
@@ -2907,26 +3039,63 @@ subroutine read_tallies_xml()
             t % score_bins(j) = N_4N
 
           case ('absorption')
+
+            if (t % find_filter(FILTER_DELAYEDGROUP) > 0) then
+              call fatal_error("Cannot tally absorption rate with a &
+                   &delayedgroup energy filter.")
+            end if
+
             t % score_bins(j) = SCORE_ABSORPTION
             if (t % find_filter(FILTER_ENERGYOUT) > 0) then
               call fatal_error("Cannot tally absorption rate with an outgoing &
                    &energy filter.")
             end if
           case ('fission')
+
+            if (t % find_filter(FILTER_DELAYEDGROUP) > 0) then
+              call fatal_error("Cannot tally fission rate with a &
+                   &delayedgroup energy filter.")
+            end if
+
             t % score_bins(j) = SCORE_FISSION
             if (t % find_filter(FILTER_ENERGYOUT) > 0) then
               call fatal_error("Cannot tally fission rate with an outgoing &
                    &energy filter.")
             end if
           case ('nu-fission')
+
+            if (t % find_filter(FILTER_DELAYEDGROUP) > 0) then
+              call fatal_error("Cannot tally nu fission rate with a &
+                   &delayedgroup energy filter.")
+            end if
+
             t % score_bins(j) = SCORE_NU_FISSION
             if (t % find_filter(FILTER_ENERGYOUT) > 0) then
               ! Set tally estimator to analog
               t % estimator = ESTIMATOR_ANALOG
             end if
+          case ('delayed-nu-fission')
+
+            t % score_bins(j) = SCORE_DELAYED_NU_FISSION
+            if (t % find_filter(FILTER_ENERGYOUT) > 0) then
+              ! Set tally estimator to analog
+              t % estimator = ESTIMATOR_ANALOG
+            end if
           case ('kappa-fission')
+
+            if (t % find_filter(FILTER_DELAYEDGROUP) > 0) then
+              call fatal_error("Cannot tally kappa fission with a &
+                   &delayedgroup energy filter.")
+            end if
+
             t % score_bins(j) = SCORE_KAPPA_FISSION
           case ('current')
+
+            if (t % find_filter(FILTER_DELAYEDGROUP) > 0) then
+              call fatal_error("Cannot tally current with a &
+                   &delayedgroup energy filter.")
+            end if
```

Comment:
A lot of redundant code here. Why not just have a check after the select case with something like,
```Fortran
if (score_name /= 'delayed-nu-fission' .and. t%find_filter(FILTER_DELAYEDGROUP) > 0) then
  ...
end if
```